index.js

```diff
@@ -1,480 +1,203 @@
-'use strict'
-var fs = require('fs')
-var request = require('request')
-var _ = require('highland')
+const axios = require('axios')
+const fs = require('fs')
+const logger = require('./logger')
+const setLogLevel = logger.setLevel
+const qs = require('qs')
+require('dotenv').config()
+
+const RETRY_ERROR = 'retry error'
+
+class RetryError extends Error {
+  constructor(type) {
+    super(type);
+    this.name = RETRY_ERROR;
+    this.message = `${type} failed after 3 attempts with empty responses`
+  }
+}
+const delay = async (time) => new Promise((resolve, reject) => setTimeout(resolve, time))
 
-var exports = module.exports = {}
-// the API creds
-exports.credsKey = null
-exports.credsSecret = null
-exports.credsBase = null
-// are we authorized against the API?
-exports.authorizedTimestamp = false
-exports.authorizedToken = null
-exports.authorizedTimer = null
 
-// see if creds are in the env
-if (process.env.SIERRA_KEY) exports.credsKey = process.env.SIERRA_KEY
-if (process.env.SIERRA_SECRET) exports.credsSecret = process.env.SIERRA_SECRET
-if (process.env.SIERRA_BASE) exports.credsBase = process.env.SIERRA_BASE
+let accessToken = null
+let credsKey = null
+let credsSecret = null
+let baseUrl = null
 
-/**
-* Loads a congig object, passed or from disk
-*
-* @param  {object|string} configOrFile - The object with the credentials or a path to a json file with the credentials
-* @return {boolean} did it load or not
-*/
-exports.loadConfig = (configOrFile) => {
-  if (typeof configOrFile === 'string') {
+if (process.env.SIERRA_KEY) credsKey = process.env.SIERRA_KEY
+if (process.env.SIERRA_SECRET) credsSecret = process.env.SIERRA_SECRET
+if (process.env.SIERRA_URL) baseUrl = process.env.SIERRA_URL
```

Comment:
Looks like you're changing `SIERRA_BASE` to `SIERRA_URL`? That may be an additional change that integrating apps will need to adapt to when using this new version of the wrapper https://github.com/search?q=SIERRA_BASE+org%3ANYPL-discovery&type=code . Was that change intentional? Looks like it's still listed as `SIERRA_BASE` in the README (but the installation test uses the newer `SIERRA_URL`). So we should settle one or the other. Keeping it as `SIERRA_BASE` is going to be easiest, but we can change it if you feel strongly.